linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala

```diff
@@ -11,6 +11,25 @@ import java.net.URLEncoder
 import java.util.Base64
 import scala.collection.breakOut
 
+/*
+  Mapping from linkerd header names to medallia header names.
+ */
+object MedalliaHeaders {
+  val MedalliaPrefix = "X-Medallia"
+  val mapHeaderNames = collection.immutable.HashMap(
+    "L5D-SAMPLE" -> "X-Medallia-Tracing-Enabled",
+    "L5D-REQID" -> "X-Medallia-Tracer-Request-Id",
+    "L5D-CTX-TRACE" -> "X-Medallia-Tracer-Span-Id"
+  )
+  def getMedalliaHeaderName(linkerdHeaderName: String): String = {
+    if (mapHeaderNames contains linkerdHeaderName.toUpperCase)
+      mapHeaderNames(linkerdHeaderName)
```

Comment:
Don't we need to toUpperCase this line too?
Let's add a simple test to avoid regressions.